crates/bevy_ecs/src/query/filter.rs

```diff
@@ -376,261 +471,393 @@ macro_rules! impl_query_filter_tuple {
+/// # struct Transform {};
+///
+/// fn print_moving_objects_system(query: Query<&Name, Changed<Transform>>) {
+///     for name in &query {
+///         println!("Entity Moved: {:?}", name);
+///     }
+/// }
+///
+/// # bevy_ecs::system::assert_is_system(print_moving_objects_system);
+/// ```
+pub struct Changed<T>(PhantomData<T>);
 
-            #[inline]
-            fn update_archetype_component_access(
-                &id: &ComponentId,
-                archetype: &Archetype,
-                access: &mut Access<ArchetypeComponentId>,
-            ) {
-                if let Some(archetype_component_id) = archetype.get_archetype_component_id(id) {
-                    access.add_read(archetype_component_id);
-                }
-            }
+#[doc(hidden)]
+#[derive(Clone)]
+pub struct ChangedFetch<'w> {
+    table_ticks: Option<ThinSlicePtr<'w, UnsafeCell<Tick>>>,
+    sparse_set: Option<&'w ComponentSparseSet>,
+    last_run: Tick,
+    this_run: Tick,
+}
 
-            fn init_state(world: &mut World) -> ComponentId {
-                world.init_component::<T>()
-            }
+/// SAFETY:
+/// `fetch` accesses a single component in a readonly way.
+/// This is sound because `update_component_access` and `update_archetype_component_access` add read access for that component and panic when appropriate.
+/// `update_component_access` adds a `With` filter for a component.
+/// This is sound because `matches_component_set` returns whether the set contains that component.
+unsafe impl<T: Component> WorldQuery for Changed<T> {
+    type Fetch<'w> = ChangedFetch<'w>;
+    type Item<'w> = bool;
+    type State = ComponentId;
 
-            fn matches_component_set(&id: &ComponentId, set_contains_id: &impl Fn(ComponentId) -> bool) -> bool {
-                set_contains_id(id)
-            }
+    #[inline]
+    unsafe fn init_fetch<'w>(
+        world: UnsafeWorldCell<'w>,
+        &id: &ComponentId,
+        last_run: Tick,
+        this_run: Tick,
+    ) -> Self::Fetch<'w> {
+        Self::Fetch::<'w> {
+            table_ticks: None,
+            sparse_set: (T::Storage::STORAGE_TYPE == StorageType::SparseSet).then(|| {
+                world
+                    .unsafe_world()
+                    .storages()
+                    .sparse_sets
+                    .get(id)
+                    .debug_checked_unwrap()
+            }),
+            last_run,
+            this_run,
         }
+    }
 
-        /// SAFETY: read-only access
-        unsafe impl<T: Component> ReadOnlyWorldQuery for $name<T> {}
+    const IS_DENSE: bool = {
+        match T::Storage::STORAGE_TYPE {
+            StorageType::Table => true,
+            StorageType::SparseSet => false,
+        }
     };
+
+    #[inline]
+    unsafe fn set_table<'w>(
+        fetch: &mut Self::Fetch<'w>,
+        &component_id: &ComponentId,
+        table: &'w Table,
+    ) {
+        fetch.table_ticks = Some(
+            (Column::get_changed_ticks_slice)(
```

Comment:
Why is Column::get_changed_ticks_slice in parenthesis here?

Reply:
It was there because I had copied it out of a macro (where that was needed). I've fixed it (and the corresponding one in `Added`)